operator/es_client.go

```diff
@@ -256,6 +257,42 @@ func (c *ESClient) setExcludeIPs(ips string) error {
 	return nil
 }
 
+// remove the podIP from Elasticsearch exclude._ip list
+func (c *ESClient) removeFromExcludeIPList(pod *v1.Pod) error {
+
+	c.mux.Lock()
+	defer c.mux.Unlock()
```

Comment:
:+1: this looks better than the Unlock() we have in excludePodIP. Could you add the defer to that method as well?

Reply:
regarding naming of the method, as it is reversing the effect of excludePodIP, I would say `undoExcludePodIP(pod *v1.Pod)`

Reply:
btw. `Cleanup` is also calling getClusterSettings() and fiddling with IP exclusion, but without locks. I assume we need locks there as well, and check if some of the code can be re-used between these three places where we're operating on IP exclusion lists.

Reply:
Updates added for all points